File: public/public-web/src/test/java/ch/difty/scipamato/publ/web/PublicPageParametersTest.java
package ch.difty.scipamato.publ.web;

import static ch.difty.scipamato.publ.web.PublicPageParameters.*;
import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class PublicPageParametersTest {

    @Test
    void assertParameters() {
        assertThat(PublicPageParameters.values()).containsExactly(SEARCH_ORDER_ID, SHOW_EXCLUDED, SHOW_NAVBAR, NUMBER,
            ISSUE, PARENT_URL);
    }

    @Test
    void assertRoleNames() {
        assertThat(SEARCH_ORDER_ID.getName()).isEqualTo("searchOrderId");
        assertThat(SHOW_EXCLUDED.getName()).isEqualTo("showExcluded");
        assertThat(SHOW_NAVBAR.getName()).isEqualTo("showNavbar");
        assertThat(NUMBER.getName()).isEqualTo("number");
        assertThat(ISSUE.getName()).isEqualTo("issue");
        assertThat(PARENT_URL.getName()).isEqualTo("parentUrl");
    }
}

File: public/public-web/src/test/kotlin/ch/difty/scipamato/publ/web/PublicPageParametersTest.kt
package ch.difty.scipamato.publ.web

import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldContainSame
import org.junit.jupiter.api.Test

internal class PublicPageParametersTest {

    @Test
    fun assertParameters() {
        PublicPageParameters.values() shouldContainSame listOf(
            PublicPageParameters.SEARCH_ORDER_ID,
            PublicPageParameters.SHOW_EXCLUDED,
            PublicPageParameters.SHOW_NAVBAR,
            PublicPageParameters.NUMBER,
            PublicPageParameters.ISSUE,
            PublicPageParameters.PARENT_URL
        )
    }

    @Test
    fun assertRoleNames() {
        PublicPageParameters.SEARCH_ORDER_ID.parameterName shouldBeEqualTo "searchOrderId"
        PublicPageParameters.SHOW_EXCLUDED.parameterName shouldBeEqualTo "showExcluded"
        PublicPageParameters.SHOW_NAVBAR.parameterName shouldBeEqualTo "showNavbar"
        PublicPageParameters.NUMBER.parameterName shouldBeEqualTo "number"
        PublicPageParameters.ISSUE.parameterName shouldBeEqualTo "issue"
        PublicPageParameters.PARENT_URL.parameterName shouldBeEqualTo "parentUrl"
    }
}
